src/Astreiko.Homework4/Program.cs

```diff
@@ -0,0 +1,375 @@
+
+                var searchId = Console.ReadLine().Trim();
+
+                var searchEvent = CheckEvent(searchId);
+
+                if (searchEvent.BoolResult)
+                {
+                    Console.WriteLine("Event found.");
+                    Console.WriteLine();
+
+                    CurrentListEvents.RemoveAll(x => x.Id == searchId);
+
+                    Console.WriteLine("Event deleted.");
+                    Console.WriteLine();
+                    checkDel = false;
+                }
+                else
+                {
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine("Event does not find. Please try again.");
+                    Console.WriteLine();
+                    Console.ResetColor();
+
+                    Console.Write("You want continious? (y - default or n)");
+
+                    if (Console.ReadLine().Trim() == "n") { checkDel = false; }
+                };
+            }
+        }
+
+        /// <summary>
+        /// Редактирование события
+        /// </summary>
+        private static void EditEvent()
+        {
+            var checkEdit = true;
+
+            while (checkEdit)
+            {
+                Console.Write("Input id for edit : ");
+
+                var searchId = Console.ReadLine().Trim();
+
+                var searchEvent = CheckEvent(searchId);
+
+                if (searchEvent.BoolResult)
+                {
+                    Console.WriteLine("Event found.");
+                    Console.WriteLine();
+
+                    Console.Write("Enter description : ");
+                    CurrentListEvents[searchEvent.NumberRow].Description = Console.ReadLine().Trim();
+
+                    var inputDate = CheckDate();
+                    CurrentListEvents[searchEvent.NumberRow].StartDate = inputDate.StartDate;
+
+                    CurrentListEvents[searchEvent.NumberRow].EndDate = inputDate.EndDate;
+
+                    Console.Write("Enter status : ");
+                    CurrentListEvents[searchEvent.NumberRow].Status = Console.ReadLine().Trim();
+
+                    Console.WriteLine("Event edited.");
+
+                    checkEdit = false;
+                }
+                else
+                {
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine("Event does not find. Please try again.");
+                    Console.WriteLine();
+                    Console.ResetColor();
+
+                    Console.Write("You want continious? (y - default or n)");
+
+                    if (Console.ReadLine().Trim() == "n") { checkEdit = false; }
+                };
+            }
+        }
+
+        /// <summary>
+        /// Прверка существования введенного идентификатора события
+        /// </summary>
+        /// <returns></returns>
+        private static (bool BoolResult, int NumberRow) CheckEvent(string searchId)
```

Comment:
Something like FindIndex() would be more natural, when -1 is returned when nothing is found. Tuples are useful when built in equality for set of values is needed or for example multiple values should be returned without introducing a class (what itself is not a good practice but sometimes reasonable)